src/waitFor.js

```diff
@@ -38,6 +39,21 @@ export default function waitFor<T>(
   });
 }
 
+export default async function waitFor<T>(
+  expectation: () => T,
+  options?: WaitForOptions
+): Promise<T> {
+  let result: T;
+
+  //$FlowFixMe: this is just too complicated for flow
+  await act(async () => {
+    result = await waitForInternal(expectation, options);
+  });
+
+  //$FlowFixMe: either we have result or `waitFor` there error
+  return result;
```

Comment:
How about:
```suggestion
  return act(() => waitForInternal(expectation, options));
```

Reply:
Unfortunately does not work, as act is not forwarding the return value from the callback.

Reply:
We need to capture it to intermediate variable so that it can be returned by the `waitFor` async wrapper

Reply:
Ok, I see it now. The error suppressed by "//$FlowFixMe: this is just too complicated for flow" comes from lacking lib defs of react-test-renderer, not Flow being dumb. Worth updating the comment

Reply:
Updated comment